giskard/llm/client/utils.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import os
+from enum import Enum
+from typing import Optional
+
+
+class ApiType(Enum):
+    openai = (0,)
+    azure = 1
+
+
+def autodetect_client():
+    if "GISKARD_SCAN_LLM_API" in os.environ:
+        try:
+            return ApiType[os.environ["GISKARD_SCAN_API"].lower()]
+        except KeyError:
+            raise ValueError(
+                f"Environment variable GISKARD_SCAN_LLM_API value is not allowed! Allowed values are: {', '.join([api.name for api in ApiType])}"
+            )
+
+    if "AZURE_OPENAI_API_KEY" in os.environ:
+        return ApiType.azure
+
+    return ApiType.openai
+
+
+def get_scan_model(api_type: ApiType) -> Optional[str]:
+    scan_model = os.environ.get("GISKARD_SCAN_LLM_MODEL")
```

Comment:
This is tricky because we refer to GPT-4 when using the various clients. That was over-optimization, could you refactor that to get rid of the model choice at the Generator/Evaluator level? This would also include the `LLMClient` interface.